packages/flutter/lib/src/widgets/single_child_scroll_view.dart

```diff
@@ -32,6 +32,128 @@ import 'scrollable.dart';
+/// centered, or otherwise positioned using the [Flex] layout model of a [Column],
+/// the following technique can be used to provide the [Column] with a minimum
+/// dimension while allowing it to shrink-wrap the contents when there isn't enough
+/// room to apply these spacing or alignment needs.
+///
+/// A [LayoutBuilder] is used to obtain the size of the viewport (implicitly via
+/// the constraints that the [SingleChildScrollView] sees, since viewports
+/// typically grow to fit their maximum height constraint). Then, inside the
+/// scroll view, a [ConstrainedBox] is used to set the minimum height of the
+/// [Column], providing the [Column] with some remaining space to allocation
+/// as specified by its [Column.mainAxisAlignment]. In this example, the
+/// children are spaced out equally, unless there's no more room, in which case
+/// they stack vertically and scroll.
+///
+/// ```dart
+/// new LayoutBuilder(
+///   builder: (BuildContext context, BoxConstraints viewportConstraints) {
+///     return SingleChildScrollView(
+///       child: new ConstrainedBox(
+///         constraints: new BoxConstraints(
+///           minHeight: viewportConstraints.maxHeight,
+///         ),
+///         child: new Column(
+///           mainAxisAlignment: MainAxisAlignment.spaceAround,
+///           children: <Widget>[
+///             new Container(
+///               // A fixed-height child.
+///               color: Colors.yellow,
+///               height: 120.0,
+///             ),
+///             new Container(
+///               // Another fixed-height child.
+///               color: Colors.green,
+///               height: 120.0,
+///             ),
+///           ],
+///         ),
+///       ),
+///     );
+///   },
+/// )
+/// ```
+///
+/// When using this technique, [Expanded] and [Flexible] are not useful, because
+/// in both cases the "available space" is infinite (since this is in a viewport).
+/// The next section describes a technique for providing a maximum height constraint.
+///
+/// ### Expanding content to fit the viewport
+///
+/// The following example builds on the previous one. In addition to providing a
+/// minimum dimension for the child [Column], an [IntrinsicHeight] widget is used
+/// to force the column to be exactly as big as its contents. This constraint
+/// combines with the [ConstrainedBox] constraints discussed previously to ensure
+/// that the column becomes either as big as viewport, or as big as the contents,
+/// whichever is biggest.
+///
+/// The widget that is to grow to fit the remaining space so provided is wrapped
+/// in an [Expanded] widget.
+///
+/// ```dart
+/// new LayoutBuilder(
+///   builder: (BuildContext context, BoxConstraints viewportConstraints) {
+///     return SingleChildScrollView(
+///       child: new ConstrainedBox(
+///         constraints: new BoxConstraints(
+///           minHeight: viewportConstraints.maxHeight,
+///         ),
+///         child: new IntrinsicHeight(
```

Comment:
OK, I'm confused by this as well. Why wouldn't it be enough to just wrap the column with `SizedBox(height: viewportConstraints.biggest.height)`?

Reply:
`viewportConstraints.biggest.height` is `viewportConstraints.maxHeight`.

You could do that, but then if your contents don't fit, they will overflow the column, which is the whole problem we're trying to avoid. :-)